apps/AppWithWearable/pubspec.yaml

```diff
@@ -85,13 +85,15 @@ dependencies:
   encrypt: ^5.0.3
   flutter_markdown: ^0.7.2+1
   file_picker: ^8.0.5
-  record: ^5.1.1
+  record: ^5.1.2
   record_android: ^1.2.2
   record_platform_interface: ^1.1.0
   record_darwin: ^1.1.0
   flutter_sound: ^9.6.0
   audio_session: ^0.1.19
   onesignal_flutter: ^5.1.2
+  flutter_background_service: ^5.0.6
+  disable_battery_optimization: ^1.1.1
```

Comment:
Check foreground.dart, I think we are already asking for that, if not, let's add this library, otherwise let's reuse this.
```
if (!await FlutterForegroundTask.isIgnoringBatteryOptimizations) {
      await FlutterForegroundTask.requestIgnoreBatteryOptimization();
    }
```